src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests.cs

```diff
@@ -824,7 +825,7 @@ class C
             driver.RunGeneratorsAndUpdateCompilation(compilation, out _, out _);
 
             Assert.Equal(2, passedIn.Length);
-            AssertEx.SetEqual(texts, passedIn);
+            Assert.Equal((IEnumerable<AdditionalText>)texts, (IEnumerable<AdditionalText>)passedIn);
```

Comment:
nit: I was surprised that these casts are needed. But if xunit can't discover the correct method of comparison here then we might as well make the boxing explicit :).

alternatively it seems like we will pick the overload if we call Equal using a type argument e.g. `Assert.Equal<AdditionalText>(texts, passedIn)`

Reply:
Honestly this is one of those places that I can never remember the correct incantations for the assert 🙃